test/e2e/upgrade_management_cluster_test.go

```diff
@@ -35,307 +24,128 @@ func upgradeManagementCluster() {
+				Expect(err).ShouldNot(HaveOccurred())
+				Expect(clusterProxy.Apply(ctx, resource, []string{"-n", clusterNamespace}...)).ShouldNot(HaveOccurred())
 			}
-			g.Expect(n).To(Equal(2))
-		}, e2eConfig.GetIntervals(specName, "wait-machine-running")...).Should(Succeed())
-
-		listBareMetalHosts(ctx, upgradeClusterClient, client.InNamespace(namespace))
-
-		By("THE MANAGEMENT CLUSTER WITH OLDER VERSION OF PROVIDERS WORKS!")
-
-		/*--------------------------------------*
-		| Upgrade Ironic and BareMetalOperator  |
-		*---------------------------------------*/
-
-		// TODO: If/when we rework the upgrade test to be able to use CAPI e2e
-		// upgrade test, these two should be included as a PreUpgrade step.
-		// If we do not go that route, we should instead refactor the whole
-		// upgradeManagementCluster function into smaller parts.
-		upgradeIronic(upgradeClusterProxy.GetClientSet())
-		upgradeBMO(upgradeClusterProxy.GetClientSet())
-
-		/*-------------------------------*
-		| Upgrade the management cluster |
-		*--------------------------------*/
-		By("Upgrading providers to the latest version available")
-		By("Get the management cluster images before upgrading")
-		printImages(upgradeClusterProxy)
-
-		Logf("Upgrade management cluster to : %v", clusterv1.GroupVersion.Version)
-		clusterctl.UpgradeManagementClusterAndWait(ctx, clusterctl.UpgradeManagementClusterAndWaitInput{
-			ClusterctlConfigPath: clusterctlConfigPath,
-			ClusterProxy:         upgradeClusterProxy,
-			Contract:             clusterv1.GroupVersion.Version,
-			LogFolder:            filepath.Join(artifactFolder, "clusters", workloadClusterName),
-		}, e2eConfig.GetIntervals(specName, "wait-controllers")...)
-
-		Logf("The target cluster upgraded!")
-
-		By("Get the management cluster images after upgrading")
-		printImages(upgradeClusterProxy)
-
-		By("Upgrade bootstrap cluster")
-		By("Get the management cluster images before upgrading")
-		printImages(bootstrapClusterProxy)
-
-		Logf("Upgrade bootstrap cluster to : %v", clusterv1.GroupVersion.Version)
-		clusterctl.UpgradeManagementClusterAndWait(ctx, clusterctl.UpgradeManagementClusterAndWaitInput{
-			ClusterctlConfigPath: clusterctlConfigPath,
-			ClusterProxy:         bootstrapClusterProxy,
-			Contract:             clusterv1.GroupVersion.Version,
-			LogFolder:            filepath.Join(artifactFolder, "clusters", bootstrapClusterProxy.GetName()),
-		}, e2eConfig.GetIntervals(specName, "wait-controllers")...)
-
-		Logf("The source cluster upgraded!")
+		}
+	}
 
-		By("Get the bootstrap cluster images after upgrading")
-		printImages(bootstrapClusterProxy)
+	// Create bmhs in the in the namespace that will host the new cluster
+	createBMH(clusterProxy, clusterNamespace, clusterName)
+	// This is exported here because we need this only when creating the workload
+	// if exported before it breaks creating the management because it uses v1beta1 templates
+	os.Setenv("PROVIDER_ID_FORMAT", "metal3://{{ ds.meta_data.uuid }}")
```

Comment:
why this is set at the end?

Reply:
we can move this to pre-init with the IP override it is the same at the end what is important is to export this after applying the first target cluster template

Reply:
or even export is in the if block `if isBootstrapProxy {`